src/formatDate.js

```diff
@@ -9,6 +9,34 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const result = Array(3).fill(0);
+  const dateArray = date.split(fromFormat.slice(-1));
+  const day = dateArray[fromFormat.indexOf('DD')];
+  const month = dateArray[fromFormat.indexOf('MM')];
+  const yearFromFormat = fromFormat.indexOf('YY') === -1 ? 'YYYY' : 'YY';
+  const year = dateArray[fromFormat.indexOf(yearFromFormat)];
+  const yearToFormat = toFormat.indexOf('YY') === -1 ? 'YYYY' : 'YY';
+  const formatedYear = format(year, yearFromFormat, yearToFormat);
+
+  result[toFormat.indexOf('DD')] = day;
+  result[toFormat.indexOf('MM')] = month;
+  result[toFormat.indexOf(yearToFormat)] = formatedYear;
+
+  return result.join(toFormat.slice(-1));
```

Comment:
Similar to the issue in line 13, using `toFormat.slice(-1)` as the delimiter in `result.join(toFormat.slice(-1))` might not correctly join the date components if the delimiter is not the last character of `toFormat`. Ensure the correct delimiter is used.